src/GafferUI/RotateHandle.cpp

```diff
@@ -152,18 +152,13 @@ Imath::Eulerf RotateHandle::rotation( const DragDropEvent &event )
 
 	float rotate = ( closestRotation( m_drag.position( event ), m_rotation ) - closestRotation( m_drag.startPosition(), 0.0f ) );
 
-	// snap to 30 degree increments
+	// snap
 	if( event.modifiers & ButtonEvent::Control )
 	{
-		float piOverSix = M_PI / 6.0;
-		rotate /= piOverSix;
-		rotate = std::round( rotate ) * piOverSix;
-	}
-
-	// precision mode
-	if( event.modifiers & ButtonEvent::Shift )
-	{
-		rotate *= 0.1;
+		// Offset such that it behaves like round not floor.
```

Comment:
Moving scaling to PlanarDrag and then needing special knowledge of the scale factor and modifiers to do snapping here feels wrong - if the two are intertwined then shouldn't they be implemented in the same place?  In conjunction with my comment above, perhaps this implies that there should be an AngularDrag built on PlanarDrag?

Reply:
(Chatted off-line) We decided that snapping and precise mode do potentially live separate lives - in that snapping can be much more context sensitive. Eg: for a translate tool, does it snap to the world grid, or relative to the items original location? Scale also handles the result of a drag differently for XYZ .vs. single-axis. It just happens that they are using the same scale factor in some of these cases. Deferring this till we make more interactive tools and have a better take on how snap may need to behave in a broader set of scenarios.